amazon-kinesis-client/src/main/java/software/amazon/kinesis/lifecycle/ProcessTask.java

```diff
@@ -215,7 +215,7 @@ private void callProcessRecords(ProcessRecordsInput input, List<KinesisClientRec
                 shardInfoId);
 
         final ProcessRecordsInput processRecordsInput = ProcessRecordsInput.builder().records(records).cacheExitTime(input.cacheExitTime()).cacheEntryTime(input.cacheEntryTime())
-                .checkpointer(recordProcessorCheckpointer).millisBehindLatest(input.millisBehindLatest()).build();
+                .isAtShardEnd(input.isAtShardEnd()).checkpointer(recordProcessorCheckpointer).millisBehindLatest(input.millisBehindLatest()).build();
```

Comment:
FYI: `ProcessRecordsInput` is annotated by [`@Builder(toBuilder = true)`](https://projectlombok.org/features/Builder). Leveraging this functionality would simplify the code, affect this desired change, and avoid future issues if a new variable is ever added to `ProcessRecordsInput`. Example:
```
final ProcessRecordsInput processRecordsInput = input.toBuilder()
        .records(records).checkpointer(recordProcessorCheckpointer).build();
```